### DISCUSSION TITLE
Two examples of a query

### DISCUSSION THREAD

**Discussion by MarcinKrawczuk** [SChAth/ScAthiop#1601] [Q&A] [answered]
So after today's enlightening talk on Zoom I've come up with an idea for two searches. Both are searches only within work records. Is it possible to have a search which would give:
1) all the work records for which CPG number is given 
2) all the work records which have personal authorship i. e. they either have "saws:isAttributedToAuthor" or they use the <author> field (which BTW I see only in old records, has it fallen out of use?)

These are not theoretical by the way, I would be genuinely interested in knowing the results of these searches. Thank you!
@PietroLiuzzo @eu-genia @thea-m

**Comment by PietroLiuzzo** [comment, marked as answer]
Thank you very much!

> 1. all the work records for which CPG number is given

This can be achieved for example by selecting the "Other Clavis ID" filter in the work list view, e.g.:
https://betamasaheft.eu/works/list?clavistype=CPG

you can do the same for any of the other values in the menu "other clavis".

> 2. all the work records which have personal authorship i. e. they either have "saws:isAttributedToAuthor" or they use the  field (which BTW I see only in old records, has it fallen out of use?)

You can see that by selecting the author filter in the work list view
<img width="469" alt="Schermata 2020-12-08 alle 09 15 51" src="https://user-images.githubusercontent.com/508012/101457591-059b1980-3936-11eb-8a40-c6e13f68476b.png">

or more explicitly requesting that type of information with this XPath query
https://betamasaheft.eu/xpath?xpath=%24config%3Acollection-rootW%2F%2Ft%3Arelation%5B%40name%3D+%27saws%3AisAttributedToAuthor%27%5D

or with this SPARQL 

```sparql
       SELECT *
WHERE {?work saws:isAttributedToAuthor ?author}
```

https://betamasaheft.eu/sparql?query=SELECT+*%0D%0AWHERE+%7B%3Fwork+saws%3AisAttributedToAuthor+%3Fauthor%7D

you may however want to include dcterms:creator

```sparql
SELECT *
WHERE {
{?work dcterms:creator ?author}
UNION {?work saws:isAttributedToAuthor ?author}
}
```


> 
> These are not theoretical by the way, I would be genuinely interested in knowing the results of these searches. Thank you!
> @PietroLiuzzo @eu-genia @thea-m

Please let me know if these do meet your requirements or you had something different in mind.

**Comment by PietroLiuzzo** [comment]
I am not sure about what you mean with "use the field". an example would help a lot!
here are the authorship encoding guidelines https://betamasaheft.eu/Guidelines/?q=author&start=6&id=work-teiHeader

**Comment by eu-genia** [comment]
I think Marcin meant the tag `<author>` which was automatically created for records inherited from the EAe
This is not used for traditional literature as we always deal with attributions and is therefore replaced by the saws relation; the tag however is well used for modern literature when works are clearly signed

**Comment by MarcinKrawczuk** [comment]
@PietroLiuzzo Thank you very much, I will run the queries as soon as I have time and will communicate if I encounter any problems. @eu-genia This is precisely what I meant, thank you for the clarification.
